src/test/java/spoon/processing/ProcessingTest.java

```diff
@@ -87,8 +87,13 @@ public void testRuntimeProcessorManager() {
 
 	@Test
 	public void testSpoonTagger() {
+		// contract: this tests the process method of the SpoonTagger class.
 		final Launcher launcher = new Launcher();
 		launcher.addProcessor("spoon.processing.SpoonTagger");
+		// we need to make sure that the spoon/Spoon.java file doesn't already exist in the system, as it will cause the assertion to be irrelevant
+		File file = new File (launcher.getModelBuilder().getSourceOutputDirectory() + "/spoon/Spoon.java");
+		// deleting the Spoon.java if it exists
```

Comment:
This is an example of a bad comment: it just restates _what_ the code does. The code already says that.

Reply:
Thanks for pointing out, I will avoid redundant comments in future

Reply:
Note that as with any rule of thumb, there are exceptions. Sometimes we simply fail to write code that is clear enough to be understood, and then we need to put a comment in to clarify. That wasn't the case here though, the code was very clear.